### DISCUSSION TITLE
Latest Release Roadmap and Design Documents

### DISCUSSION THREAD

**Discussion by ananya-jain1** [Sunbird-cQube/community#760] [Announcements]
Hey folks,
At the onset of this year, we are delighted to share a few updates regarding cQube with all of you:

🚀 **1. Release roadmap - [Link](https://project-sunbird.atlassian.net/jira/software/c/projects/CQ/boards/117/roadmap)**

The roadmap encompasses the plan for upcoming releases of cQube. cQube v5.0 is aimed to launch in March 2023 with revamped installation, ingestion, processing and visualisation blocks, where the datasets formed can be connected with any external visualisation tool.

Any other features required by the community are welcome in the roadmap! Feel free to open a discussion forum regarding any new ideas or feature requests. 💡

In order to make these releases happen, we organise regular demos. You can have a look at the previous demos [here](https://drive.google.com/drive/folders/1QUYDLHo0u1onJIjOHy2cSeiUMpEKLRW0). For all those who wish to join, next demo is scheduled on 16th March, 2023 at 3:30 pm IST on [this](https://us06web.zoom.us/j/5987446879) link.

Following is a draft list of programs planned to be made live in v5.0 Beta and v5.0:

#### **v5.0 Beta:**
- School Statistics
- Attendance
- Student Assessment
- Teacher Mentoring
- Review & Monitoring
- Nipun Bharat Mission

#### **v5.0:**
- Mid-day Meals
- Teacher Training
- Resource Distribution
- School Infrastructure
- Out of School Children
- National Programs
- District Scorecard

_The draft list may be updated based on consultations with states and other potential adopters._

📑 **2. HLD and LLDs - [HLD Link](https://project-sunbird.atlassian.net/wiki/spaces/CQUB/pages/3258024044?atlOrigin=eyJpIjoiOTc0NThlMGNkMDg1NGQ4Mzk4OTZlMzE2ZDRiNjQ3ZTAiLCJwIjoiaiJ9) ; [LLD Link](https://project-sunbird.atlassian.net/wiki/spaces/CQUB/pages/3266740614/cQube+%7C+LLD+-+v12Dec2022?atlOrigin=eyJpIjoiZGRjYzMxMTQ0MDgzNDUzNGI4ZWE2OWM0MjkyZmE0YmQiLCJwIjoiaiJ9)**

HLD document details the high-level design of all blocks being built as part of cQube roadmap. LLD Document explains the designs for ingestion and processing blocks being released in cQube v5.0 Beta in February.

We'd happy to take suggestions regarding any design decisions and approach detailed out in the documents mentioned above. Feel free to open a discussion forum regarding any comments. 💬

🗂️ **3. Backlog Summary - [Link](https://github.com/Sunbird-cQube/community/issues)**

There are a total of 148 open issues in the community repository. Following is a breakup of these open issues:

- Being addressed in v5.0 Beta (by February): 34
- Being addressed in v5.0 (by March): 42
- Being addressed in v5.1 (by April - May): 33
- Need description by issue author: 17
- Need inputs from technical design owners: 4

_18 open issues are pending for analysis and the statistics will be updated once this is done._

👥 **4. Areas for community contribution**

While analysing all the open issues, we identified a few areas that are fit to be contributed by the community:

- Experiment of using Spark as a processor instead of NiFi
- Enabling a config in ingestion with an option to select if the header is present in the files being uploaded. If the header is present, the order of the columns is not important, else it is.

We are looking forward to more and more community contributions to cQube. Any suggestions around areas where this can be enabled are appreciated!

For all interested folks, we are currently experimenting on explaining the journey of an event once it is ingested into cQube based on 2 parameters - granularity and frequency. It covers use cases regarding if data being sent as events requires aggregation or not and how processing will change if the frequency of data ingested is a day, a week, a month or a year etc. You can follow this PoC [here](https://github.com/ChakshuGautam/cQube-POCs/tree/master/docs).

**Comment by ananya-jain1** [comment]
@santhoshvasabhaktula

**Comment by ananya-jain1** [comment]
Hey folks,
The latest recorded demo for cQube v5.0 Beta can be accessed from the cQube Demo repository linked [here](https://drive.google.com/drive/folders/1QUYDLHo0u1onJIjOHy2cSeiUMpEKLRW0?usp=share_link). It showcases the following functionalities:

- One-step installation
- Step-by-Step ingestion through CSVs for Attendance Program
- Creation of datasets
- Connection of datasets to Metabase
- Creation of visualisations on Metabase

In the current sprint, we are working on some final enhancements and end-to-end testing which can be seen on the sprint plan [here](https://project-sunbird.atlassian.net/jira/software/c/projects/CQ/boards/117/backlog?selectedIssue=CQ-184&issueLimit=100).

Looking forward to any thoughts, questions or comments!

@sukhpreetsamagra @sagarihanda @surendrasinghs @SanthoshVasabhaktula @ChakshuGautam @Shruti3004

**Comment by ananya-jain1** [comment]
Hey folks,
With cQube v5.0 launching very soon 🚀🚀, I am sharing the updated roadmap with detailed functionalities [here](https://docs.google.com/spreadsheets/d/1e8b_kLCfD0Oce9Jek4nrpBFQRk-IJOKgYy5GOnFs8Ho/edit#gid=1670621804). It also consists of the versioning of various programs being made live in cQube. 

Feel free to comment on the sheet in case of any inputs or recommendations.

Stay tuned for more updates regarding the launch of cQube v5.0! 📣

@surendrasinghs @sukhpreetsamagra @sagarihanda @ChakshuGautam @SanthoshVasabhaktula

**Reply by sagarihanda** [reply]
@karsood
